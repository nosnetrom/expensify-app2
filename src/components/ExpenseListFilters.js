import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import {
  setTextFilter,
  sortByDate,
  sortByAmount,
  setStartDate,
  setEndDate,
} from '../actions/filters';

export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null,
  };
  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };
  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  };
  onTextChange = (event) => {
    this.props.setTextFilter(event.target.value);
  };
  onSortChange = (event) => {
    event.target.value == 'date'
      ? this.props.sortByDate()
      : this.props.sortByAmount();
  };
  render() {
    return (
      <div className="ExpenseListFiltersWrapper">
        <section>
          <label htmlFor="txtFilterBy">Filter by </label>
          <input
            type="text"
            value={this.props.filters.text}
            name="txtFilterBy"
            placeholder="Enter filter word"
            onChange={this.onTextChange}
          />
        </section>
        <section>
          <label htmlFor="ddlSortBy">Sort by </label>
          <select
            name="ddlSortBy"
            value={this.props.filters.sortBy}
            onChange={this.onSortChange}
          >
            <option value="date">Date</option>
            <option value="amount">Amount</option>
          </select>
        </section>
        <section>
          <label htmlFor="startDate">Dates </label>
          <DateRangePicker
            startDate={this.props.filters.startDate}
            endDate={this.props.filters.endDate}
            onDatesChange={this.onDatesChange}
            focusedInput={this.state.calendarFocused}
            onFocusChange={this.onFocusChange}
            regular="true"
            numberOfMonths="1"
            isOutsideRange={() => false}
            showClearDates="true"
          />
        </section>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  filters: state.filters,
});

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount()),
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
