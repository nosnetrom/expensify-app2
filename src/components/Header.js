import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header>
    <nav>
      <ul>
        <li>
          <NavLink to="/" exact activeClassName="activeNavLink">
            Home
          </NavLink>
        </li>
        <li>
          <NavLink to="/add" exact activeClassName="activeNavLink">
            Add Expense
          </NavLink>
        </li>
        <li>
          <NavLink to="/help" exact activeClassName="activeNavLink">
            Help
          </NavLink>
        </li>
      </ul>
    </nav>
    <h1>Expensify!</h1>
  </header>
);

export default Header;
