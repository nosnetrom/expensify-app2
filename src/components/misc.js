import moment from 'moment';

// Miscellaneous helper functions of my own: JHM

export const amountProcessor = (fullAmt) => {
    let extension = '';
    const amt = parseInt(fullAmt) / 100;
    if (fullAmt % 100 == 0) {
      extension = '.00';
    } else if (fullAmt % 10 == 0) {
      extension = '0';
    }
    return `$${amt}${extension}`;
}

export const friendlyDate = (d) => {
    const fd = moment(d).format('DD MMM YYYY');
    return fd;
}
