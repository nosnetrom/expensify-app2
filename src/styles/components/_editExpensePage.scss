@import '../base/base';

.editExpenseRouteWrapper {
  .btnDeleteItem {
    background-color: $stop-color;
    border: none;
    border-radius: $my-font-size;
    color: $txt-color;
    display: block;
    font-size: $my-font-size;
    font-weight: 600;
    margin: auto auto;
    padding: 0.5 * $my-font-size;
    width: 100%;
  }
}
