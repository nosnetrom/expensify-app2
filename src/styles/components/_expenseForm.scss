@import '../base/base';

.expenseFormWrapper form {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: baseline;

  section {
    width: 23%;
    display: block;
    label,
    input {
      display: block;
      width: 100%;
    }
    label {
      color: saturate(lighten($highlight-color, 10%), 10%);
      font-weight: 600;
      padding-bottom: 0.166 * $my-font-size;
    }
    input {
      padding: 0.5 * $my-font-size;
      font-size: 1.2 * $my-font-size;
    }
  }
  #btnSubmit {
    background-color: $highlight-color;
    border: none;
    border-radius: $my-font-size;
    color: $txt-color;
    font-size: $my-font-size;
    font-weight: 600;
    width: 100%;
  }
}

.SingleDatePicker {
  width: 100%;

  .SingleDatePickerInput {
    border-radius: 0.33 * $my-font-size;
  }

  .DateInput__display-text {
    background-color: transparent !important;
    color: initial;
    font-size: 1.2 * $my-font-size;
    line-height: initial;
    padding: 0;
    width: 100%;
  }
}

@media (max-width: 1279px) {
  .expenseFormWrapper form {
    section {
      width: 48%;
      padding-bottom: 0.5 * $my-font-size;
    }
  }
}
