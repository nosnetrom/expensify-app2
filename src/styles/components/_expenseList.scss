@import '../base/base';

table.expenseListWrapper {
  width: 90%;
  margin: $my-font-size auto;
  border-collapse: separate;
}
th {
  background-color: $highlight-color;
  text-shadow: 0.5px 0.5px 0 $app-bg-color;
}
