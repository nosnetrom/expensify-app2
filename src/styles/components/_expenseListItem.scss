@import '../base/base';

td {
  vertical-align: top;
  color: $bg-color;
  background-color: $txt-color;
  padding: 0.166rem 0.33rem;
}
td[colspan='3'] {
  padding: 0.33 * $my-font-size;
  text-align: center;
}
.itemDescription {
  font-weight: 600;
  padding: 0.33 * $my-font-size;
  a {
    color: $bg-color;
    text-decoration: none;
    border-bottom: dotted 1px;
  }
}
.itemAmount,
.itemCreatedAt {
  padding: 0.33 * $my-font-size;
  text-align: right;
}
.btnDeleteItem {
  background-color: $stop-color;
  border: none;
  border-radius: 50%;
  color: $txt-color;
  display: inline;
  font-size: 0.67 * $my-font-size;
  font-weight: 900;
  margin: 0.25 * $my-font-size 0.5 * $my-font-size 0.166 * $my-font-size 0.166 *
    $my-font-size;
  padding: 0.166 * $my-font-size 0.33 * $my-font-size;
  vertical-align: middle;
}
