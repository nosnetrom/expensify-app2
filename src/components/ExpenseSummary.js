import React from 'react';
import { connect } from 'react-redux';
import { amountProcessor } from './misc';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = ({expenseCount, expenseTotal}) => {
    const expenseSuffix = expenseCount === 1 ? '' : 's';
    const expenseTotalFormatted = amountProcessor(expenseTotal);
    return (
        <div className="ExpenseSummaryWrapper">
            <h2>
                Viewing {expenseCount} expense{expenseSuffix} totalling {expenseTotalFormatted}
            </h2>
        </div>
    )
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpenseSummary);
