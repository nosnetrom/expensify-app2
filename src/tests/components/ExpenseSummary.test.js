import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';

test('should correctly render ExpensesSummary with a single expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount = {1} expenseTotal = {1498} />);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount = {2} expenseTotal = {1498} />);
    expect(wrapper).toMatchSnapshot();
});
