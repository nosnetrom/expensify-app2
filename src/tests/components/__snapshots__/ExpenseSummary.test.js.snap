// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with a single expense 1`] = `
<div
  className="ExpenseSummaryWrapper"
>
  <h2>
    Viewing 
    1
     expense
     totalling 
    $14.98
  </h2>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<div
  className="ExpenseSummaryWrapper"
>
  <h2>
    Viewing 
    2
     expense
    s
     totalling 
    $14.98
  </h2>
</div>
`;
